helm/templates/_config.tpl

```diff
@@ -0,0 +1,175 @@
+{{- define "ucentralgw.config" -}}
+#
+# uCentral protocol server for devices. This is where you point
+# all your devices. You can replace the * for address by the specific
+# address of one of your interfaces
+#
+ucentral.websocket.host.0.backlog = {{ .Values.public_env_variables.UCENTRAL_WEBSOCKET_HOST_0_BACKLOG }}
```

Comment:
Can't we automate this logic by automatically converting something like `UCENTRAL_STORAGE_TYPE_POSTGRESQL_USERNAME` to `storage.type.postgresql.port`?
Otherwise this is going to be a huge pain to maintain.

Reply:
We could, but there is a `logging.channels.c2.purgeCount`configuration parameter that kinda breaks all the logic. I think we could change this parameter ourself and it would be OK

Reply:
@dunaev-opsfleet I don't follow... I see nothing special about this configuration parameter? as @brennerm has suggested, lets simplify by just turning the config template into a hash, but we should avoid naming our own inderict parameters in the middle.

Reply:
@leonidlm OK, let me give you more examples
Say, we have env variable  `UCENTRAL_SYSTEM_DATA` which is mapped to parameter `ucentral.system.data`. This convertion is quite easy - make all text lowered and then substitute dots to underscore
But following this logic we have variable `UCENTRAL_LOGGING_CHANNELS_C2_PURGECOUNT` which would be converted to `logging.channels.c2.purgecount` which is different from the real configuration parameter - `logging.channels.c2.purgeCount` (pay attention to C in Count)
Based on code, I can't see how to change that unfortunately

Reply:
@dunaev-opsfleet What if we use a standard YAML dict instead of uppercase env variables? Something like:
```
config:
  ucentral:
    system:
      data: ...
logging:
  channels:
    c2:
      purgeCount: ...
```

Reply:
I've simplified the generation method, so at the moment the only thing we have in diff is the configuration parameter:
```
16c16
<     logging.channels.c2.purgecount = 20
---
>     logging.channels.c2.purgeCount = 20
60d59
<     ucentral.config = /ucentral-data
88d86
<     ucentral.root = /ucentral-data
```

One of the workarounds is really do it the way you've suggested but problem with this is that for one parameter we would need to use 3-5 lines which seems like overkill
Another solution is that we could use method for transformation - `swapcase`. With this we can call environment variable as `LOGGING_CHANNELS_C2_PURGEcOUNT` and it should do the trick even though it would look a bit strange

Reply:
Mmh why are all config parameters declared as uppercase `public_env_variables` anyway? Seems pretty confusing to me.

Reply:
I've made it this way since it's a good practice to use uppercase for environment variables names. But since you've mentioned it, maybe we really could simply use environment variables with same case as in configuration file. How does that sound to you?

Reply:
I'm aware of the uppercase convention for environment variables but I'm wondering why these config parameters are declared as environment variables at all. Helm simply uses them to generate the `ucentral.properties` file so why put them under `public_env_variables` and not something like `properties` in the `values.yaml` file.

Reply:
I was thinking about maybe using that as a base for a future task with environment variables, but now I understand your logic that it would be more clear if we would not be linking to environment variables
Let me redo this

Reply:
@brennerm could you please check current variant of implementation?

Reply:
Looking good :+1: